ui-charts/src/spaceTimeChart/components/TimeCaptions.tsx

```diff
@@ -72,71 +87,91 @@ const TimeCaptions = () => {
+        ...mark,
+        time: +t,
+      }));
+      if (!hideDates)
+        allMarksArray = allMarksArray.concat(
+          map(
+            computeVisibleTimeMarkers(minT, maxT, [24 * HOUR], [1], (level: number) => ({
+              level,
+              styles: dateCaptionsStyle,
+              formatter: DATES_FORMATER,
+            })),
+            (mark, t) => ({
+              ...mark,
+              time: +t,
+            })
+          )
+        );
       // Render caption background:
       ctx.fillStyle = background;
       if (!swapAxis) {
-        ctx.fillRect(0, spaceAxisSize - CAPTION_SIZE, timeAxisSize, CAPTION_SIZE);
+        ctx.fillRect(0, spaceAxisSize - captionSize, timeAxisSize, captionSize);
       } else {
-        ctx.fillRect(0, 0, CAPTION_SIZE, timeAxisSize);
-      }
-
-      // Render caption top border:
-      ctx.strokeStyle = timeGraduationsStyles[1].color;
-      ctx.lineWidth = timeGraduationsStyles[1].width;
-      if (!showTicks) {
-        ctx.beginPath();
-        if (!swapAxis) {
-          const y = getCrispLineCoordinate(spaceAxisSize - CAPTION_SIZE, ctx.lineWidth);
-          ctx.moveTo(0, y);
-          ctx.lineTo(timeAxisSize, y);
-        } else {
-          const x = getCrispLineCoordinate(CAPTION_SIZE, ctx.lineWidth);
-          ctx.moveTo(x, 0);
-          ctx.lineTo(x, timeAxisSize);
-        }
-        ctx.stroke();
+        ctx.fillRect(0, 0, captionSize, timeAxisSize);
       }
 
       // Render time captions:
-      for (const t in labelMarks) {
-        const { level, rangeIndex } = labelMarks[t];
-        const styles = timeCaptionsStyles[level];
-        const formatter = RANGES_FORMATER[rangeIndex];
-        const text = formatter(+t, pixelsPerMinute);
+      allMarksArray.forEach(({ styles, formatter, time }) => {
+        const text = formatter(time, pixelsPerMinute);
 
-        ctx.textAlign = 'center';
+        ctx.textAlign = styles.textAlign || 'center';
         ctx.textBaseline = 'top';
         ctx.fillStyle = styles.color;
+        ctx.lineWidth = 5;
+        ctx.strokeStyle = background;
+        ctx.lineCap = 'butt';
         ctx.font = `${styles.fontWeight || 'normal'} ${styles.font}`;
-        const timePixel = getCrispLineCoordinate(getTimePixel(+t), ctx.lineWidth);
+        const timePixel = getCrispLineCoordinate(getTimePixel(time), ctx.lineWidth);
 
         if (!swapAxis) {
           if (showTicks) {
             ctx.strokeStyle = timeCaptionsStyles[1].color;
-            ctx.moveTo(timePixel, spaceAxisSize - CAPTION_SIZE);
-            ctx.lineTo(timePixel, +t % 180000 === 0 ? 8 : 4);
+            ctx.moveTo(timePixel, spaceAxisSize - captionSize);
+            ctx.lineTo(timePixel, time % 180000 === 0 ? 8 : 4);
             ctx.stroke();
           }
-          ctx.fillText(text, timePixel, spaceAxisSize - CAPTION_SIZE + (styles.topOffset || 0));
+
+          ctx.strokeText(text, timePixel, spaceAxisSize - captionSize + (styles.topOffset || 0));
```

Comment:
Hum I don't get why you added a strokeText here, I don't see any difference on the storybook (but maybe I'm missing smth)